src/language-js/embed.js

```diff
@@ -284,31 +285,42 @@ function replacePlaceholders(quasisDoc, expressionDocs) {
         .concat([at + placeholder])
         .concat(rest);
     }
-    const atPlaceholderIndex = parts.findIndex(
-      (part) =>
-        typeof part === "string" && part.startsWith("@prettier-placeholder")
-    );
-    if (atPlaceholderIndex > -1) {
-      const placeholder = parts[atPlaceholderIndex];
-      const rest = parts.slice(atPlaceholderIndex + 1);
-      const placeholderMatch = placeholder.match(
-        /@prettier-placeholder-(.+)-id([\S\s]*)/
-      );
-      const placeholderID = placeholderMatch[1];
-      // When the expression has a suffix appended, like:
-      // animation: linear ${time}s ease-out;
-      const suffix = placeholderMatch[2];
-      const expression = expressions[placeholderID];
 
-      replaceCounter++;
-      parts = parts
-        .slice(0, atPlaceholderIndex)
-        .concat(["${", expression, "}" + suffix])
-        .concat(rest);
-    }
+    let tokensCount = 0;
+    parts.slice().forEach((part, idx) => {
+      if (typeof part !== "string" || !part.includes("@prettier-placeholder")) {
+        return;
+      }
+
+      // When we have multiple placeholders in one line, like:
+      // ${Child}${Child2}:not(:first-child)
+      const placeholders = part.split("@prettier-placeholder");
+      const tokens = [];
+      placeholders.forEach((placeholder, idx) => {
+        if (idx === 0) {
+          // The first item is either empty or the prefix not an actual placeholder
+          tokens.push(placeholder);
+          return;
+        }
+
+        placeholder = "@prettier-placeholder" + placeholder;
+        const placeholderMatch = placeholder.match(
+          /@prettier-placeholder-(\d+)-id([\S\s]*)/
+        );
+        const placeholderID = placeholderMatch[1];
+        // When the expression has a suffix appended, like:
+        // animation: linear ${time}s ease-out;
+        const suffix = placeholderMatch[2];
+        const expression = expressions[placeholderID];
+        tokens.push("${", expression, "}" + suffix);
+        replaceCounter++;
+      });
+
+      parts.splice(idx + tokensCount, 1, ...tokens);
```

Comment:
Overall looks good, but the logic can be simplified. Instead of modifying (`splice`ing) the same array repetitively, it would be more straightforward to create an empty array first and add elements to it, wouldn't it?

Reply:
Done.